### DISCUSSION TITLE
Easily replace builtin JWT authentication with IDP?

### DISCUSSION THREAD

**Discussion by megastary** [loco-rs/loco#275] [Q&A]
Hello,
did any of you try to replace loco's default user authentication with some other IDP via OIDC? For example I am interested in authenticating against AAD, Authentik or Zitadel.
If not, where should I try to implement it without breaking the project?
Thanks!

**Comment by jondot** [comment]
Hi,
When you generate a new project, you have an auth controller (`controllers/auth.rs`). You can copy it, and start making modifications that you need.
